flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1182,6 +1206,7 @@ def teardown_example(self, token):
         """
         Cleanup after running a hypothesis example.
         """
+        umount_all(self.deployer.mountroot)
```

Comment:
So I realized that the reason umount_all had that retry_failure nonsence was to work around situations where we were umounting by blockdevice and the blockdevices were mounted to the same mountpoint. Now that we are umounting by mountpoint we don't need the retry logic. This made it trivial to make the function synchronous.